src/analyses/ai_storage.h

```diff
@@ -103,7 +178,18 @@ class location_sensitive_storaget : public ai_storage_baset
     return it->second;
   }
 
-  statet &get_state(locationt l, const ai_domain_factory_baset &fac) override
+  statet &get_state(trace_ptrt p, const ai_domain_factory_baset &fac) override
+  {
+    register_trace(p);
+    return get_state(p->current_location(), fac);
+  }
+
+  // For backwards compatability
+  // Care should be exercised in using this.  It is possible to create domains
+  // without the any corresponding history.  This can lead to somewhat
```

Comment:
-> "without the corresponding history."

Reply:
"It is possible to create domains without any corresponding history object(s)"